dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -604,5 +622,120 @@ public String getUrl(Contentlet contentlet) {
 
     }
 
+    @Test
+    @UseDataProvider("listSerializeTestCases")
+    public void Test_Serialize_Contentlet_Then_Recover_Then_Transform(final SerializationTestCase serializationTestCase)
+            throws IOException, ClassNotFoundException, DotDataException {
+
+        final File file = new File(directory,String.format("%s.serialized",System.currentTimeMillis()));
+        file.createNewFile();
+
+        final Contentlet original = serializationTestCase.contentlet;
+        serializeContentlet(serializationTestCase.contentlet, file);
+        final Contentlet copy = readSerializedContentlet(file);
+        validateTransformation(original, copy);
+        final AssertionStrategy assertionStrategy = serializationTestCase.assertionStrategy;
+        if(null != assertionStrategy) {
+            assertionStrategy.apply(original, copy);
+        }
+    }
+
+    private static class SerializationTestCase {
+
+        final Contentlet contentlet;
+        final AssertionStrategy assertionStrategy;
+
+        SerializationTestCase(final Contentlet contentlet,
+                final AssertionStrategy assertionStrategy) {
+            this.contentlet = contentlet;
+            this.assertionStrategy = assertionStrategy;
+        }
+    }
+
+    @DataProvider
+    public static Object[] listSerializeTestCases() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5883765)